### ISSUE TITLE
This polyfill fails to load in FF 45.9.0 on Mac OS 10.13. FF47 is ok.

### ISSUE THREAD

**Issue by thoov** [thoov/ember-weakmap#21]
This is an exploratory issue for the discussions started in #19.

**Comment by thoov** [comment]
@rwjblue @jviney 

Actually I found something really weird:

Running this on firefox 45 will work:

```html
<html>
  <head>
    <meta charset="UTF-8"> 
  </head>
  <body>
    <script>
      (function() {      
        if (true) {
          var foo = foobar();
    
          function foobar() {
            return 'bazbar';
          }

          console.log(foo);
        }
      })();
    </script>
  </body>
</html>
```

But if you remove the if statement it does not work:

```html
<html>
  <head>
    <meta charset="UTF-8"> 
  </head>
  <body>
    <script>
      (function() { 
          var foo = foobar();
    
          function foobar() {
            return 'bazbar';
          }

          console.log(foo);
      })();
    </script>
  </body>
</html>
```

I dont understand why this would cause firefox to not hoist.

**Comment by thoov** [comment]
Apparently: http://statichtml.com/2011/spidermonkey-function-hoisting.html

**Comment by thoov** [comment]
So to full close this circle: The issue is that by the spec JS does not like nested function declarations within block statements and Firefox complies with that spec where other browser handle it differently. @jviney's issue surfaced because in Firefox 45 the polyfill is included (and the hoisted function was undefined) where as in Firefox 46+ the polyfill is not included (thus this hoisting problem is not included).

**Comment by jviney** [comment]
It all makes sense now, thanks for looking into it further. Can we get a point release with the fix in it?

**Comment by thoov** [comment]
@jviney I have released v3.1.1 with these changes and will send a PR into ember-macro-helpers to update WeakMap

**Comment by jviney** [comment]
Perfect, thanks.
